File: Inverted-Index/src/tests/MapperTest.java
package tests;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

import org.junit.Test;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;

import file_handler.Doc;
import search.Mapper;

public class MapperTest {
    private final String WORD1 = "word1";
    private final String WORD2 = "word2";
    private final String WORD3 = "word3";

    private final Doc DOC_TEST1;
    private final Doc DOC_TEST2;

    private Set<Doc> set1;
    private Set<Doc> set2;
    private Set<Doc> set12;

    public MapperTest() {
        DOC_TEST1 = new Doc("1", Arrays.asList(WORD1, WORD2, WORD3));
        DOC_TEST2 = new Doc("2", new ArrayList<>());
    }

    @Before
    public void initSets() {

        set1 = new HashSet<>();
        set2 = new HashSet<>();
        set12 = new HashSet<>();

        set1.add(DOC_TEST1);

        set2.add(DOC_TEST2);

        set12.add(DOC_TEST1);
        set12.add(DOC_TEST2);
    }

    @Test
    public void mergeMapsTest() {

        var map1 = new HashMap<String, Set<Doc>>();
        var map2 = new HashMap<String, Set<Doc>>();
        var map3 = new HashMap<String, Set<Doc>>();

        map1.put(WORD1, set1);
        map1.put(WORD2, set2);

        map2.put(WORD1, set2);
        map2.put(WORD3, set1);

        map3.put(WORD1, set12);
        map3.put(WORD2, set2);
        map3.put(WORD3, set1);

        var mapper = new Mapper();
        var result = mapper.mergeMaps(map1, map2);
        
        Assert.assertEquals(map3, result);
    }

    @Test 
    public void getDocMapTest() {
        var mapper = new Mapper();
        var map = mapper.getDocMap(DOC_TEST1);

        Assert.assertEquals(map.get(WORD1), set1);
    }

    @After
    public void dispose() {
        set1 = null;
        set2 = null;
        set12 = null;
    }
}

File: Phase5-TDD/InvertedIndex.Test/MapperTest.cs
using System;
using System.Collections.Generic;
using Xunit;
using InvertedIndex.Search;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Test
{
    public class MapperTest : IDisposable
    {
        private const string WORD1 = "word1";
        private const string WORD2 = "word2";
        private const string WORD3 = "word3";

        private readonly Doc DOC_TEST1;
        private readonly Doc DOC_TEST2;

        private HashSet<Doc> _set1;
        private HashSet<Doc> _set2;
        private HashSet<Doc> _set12;

        private Mapper _mapper;

        public MapperTest()
        {
            DOC_TEST1 = new Doc("1", new List<string> () {WORD1, WORD2, WORD3});
            DOC_TEST2 = new Doc("2", new List<string>());

            _set1 = new HashSet<Doc>();
            _set2 = new HashSet<Doc>();
            _set12 = new HashSet<Doc>();

            _set1.Add(DOC_TEST1);
            _set2.Add(DOC_TEST2);
            _set12.Add(DOC_TEST1);
            _set12.Add(DOC_TEST2);

            _mapper = new Mapper();
        }

        [Fact]
        public void MergeMapsTest()
        {
            var map1 = new Dictionary<string, HashSet<Doc>>();
            var map2 = new Dictionary<string, HashSet<Doc>>();
            var map3 = new Dictionary<string, HashSet<Doc>>();

            map1.Add(WORD1, _set1);
            map1.Add(WORD2, _set2);

            map2.Add(WORD1, _set2);
            map2.Add(WORD3, _set1);

            map3.Add(WORD1, _set12);
            map3.Add(WORD2, _set2);
            map3.Add(WORD3, _set1);

            var result = _mapper.MergeMaps(map1, map2);
            
            Assert.Equal(map3, result);
        }

        [Fact]
        public void GetDocMapTest()
        {
            var map = _mapper.GetDocMap(DOC_TEST1);

            Assert.Equal(map[WORD1], _set1);
        }

        public void Dispose()
        {
            GC.Collect();
        }
    }
}